### PR TITLE
Introduce tree algorithm

### PR DESCRIPTION
**Pull request by letmaik**
As discussed, this introduces a top-level tree algorithm to fork the format and makes the existing description match CCF 2.0. There may be scope to move the `LeafEntry` structure and associated "COSE_Sign1 Countersign type" section to a common section since that seems independent of any Merkle-related algorithms.

@fournet @ad-l

### CODE REVIEW

File: draft-birkholz-scitt-receipts.md [hunk 1]
```diff
@@ -60,13 +60,55 @@ This document defines a method for countersigning of COSE_Sign1 messages when us
 
 {: #mybody}
 
-# Cryptographic Components
+# Common Parameters
 
-## Merkle Trees
+A Transparency Service backed by a Merkle tree is defined by a collection of immutable parameters, which are used by clients to communicate with the service and to verify artifacts. Each of these parameters must be established before the service operator begins to operate the service. At minimum, the following must be defined:
```
Comment (henkbirkholz) [inline draft-birkholz-scitt-receipts.md:65]: communicate = issuing/adding and verifying?

Comment (letmaik) [reply draft-birkholz-scitt-receipts.md:65]: That reflects my common struggle on having two separate documents with API not being in here. But yes.

File: draft-birkholz-scitt-receipts.md [hunk 2]
```diff
@@ -252,57 +251,49 @@ ProofElement = [
 ]
 
 LeafInfo = [
-    prefix: bstr,
-    type: LeafEntryType,
-    data: LeafReceiptData
+    writeset_hash: bstr,
```
Comment (achamayou) [inline draft-birkholz-scitt-receipts.md]: Asking out of curiosity more than anything else, but this is "write_set_digest" in CCF, is there a reason why it's been renamed?

File: draft-birkholz-scitt-receipts.md [hunk 3]
```diff
@@ -252,57 +251,49 @@ ProofElement = [
...
 
-2. Let ROOT_HASH be the root hash of the Merkle tree that contains LEAF and SIGNATURE the signature over this root.
+2. Let ROOT_HASH be the root hash of the Merkle tree that contains LEAF, and SIGNATURE the signature over this root.
 
 3. Compute LEAF_HASH as the hash of LEAF.
 
 4. Generate an inclusion proof from LEAF_HASH to ROOT_HASH.
 
-5. Construct a LeafInfo structure with the implementation-specific prefix, the LeafEntryType, and the LeafReceiptData using the procedure defined by the leaf entry type.
+5. Construct a LeafInfo structure with the CCF writeset digest, the commit evidence, and the protected header parameters of the countersigner.
```
Comment (achamayou) [inline draft-birkholz-scitt-receipts.md]: Not being a native speaker, I have no opinion on the "write set" vs "writeset" distinction, but I think sticking to either "digest" or "hash" (see above) throughout would be more consistent.

**Review by achamayou — APPROVED** [review]

**Pull request merged**